tests/functional/monitoring/prometheus/alerts/test_alerting_works.py

```diff
@@ -39,17 +40,48 @@ def test_alerting_works(threading_lock):
 def test_prometheus_rule_failures(threading_lock):
     """
     There should be no PrometheusRuleFailures alert when OCS is configured.
+    This test is extended to check for many-to-many matching errors in Prometheus logs.
+    If such error message found PrometheusRuleFailures alert must fire as well.
     """
+    # any check with state False will fail the test
+    test_results = {}
+
     prometheus = ocs_ci.utility.prometheus.PrometheusAPI(threading_lock=threading_lock)
     alerts_response = prometheus.get(
         "alerts", payload={"silenced": False, "inhibited": False}
     )
-    assert alerts_response.ok is True
+    test_results["alert-msg-ok-check"] = alerts_response.ok is True
     alerts = alerts_response.json()["data"]["alerts"]
     log.info(f"Prometheus Alerts: {alerts}")
-    assert constants.ALERT_PROMETHEUSRULEFAILURES not in [
-        alert["labels"]["alertname"] for alert in alerts
-    ]
+    test_results[f"{constants.ALERT_PROMETHEUSRULEFAILURES}-present-check"] = (
+        constants.ALERT_PROMETHEUSRULEFAILURES
+        not in [alert["labels"]["alertname"] for alert in alerts]
+    )
+    prometheus_pods = get_pod_name_by_pattern(
+        "prometheus-k8s", constants.MONITORING_NAMESPACE
+    )
+    for pod_name in prometheus_pods:
+        log.info(f"Checking logs of pod {pod_name}")
+        pod_logs = get_pod_logs(
+            pod_name=pod_name,
+            namespace=constants.MONITORING_NAMESPACE,
+        ).splitlines()
+        pod_logs.reverse()
+        for log_line in pod_logs:
+            if "many-to-many matching not allowed" in log_line.lower():
+                log.error(
+                    f"Found 'warn' msg in logs of pod {pod_name}:"
+                    f"\n\n****************** Log message: ******************\n{log_line}"
+                    f"\n****************** End of logs ******************"
+                    f"\n\nThis is a known issue. Follow OCSQECL-5384 and knowledge base article "
```

Comment:
OCSQECL-5384? instead should be add DFBUG id?

Reply:
Why? 
This test improvement is a part of Corrective measures and Escape analysis initiative. It has link to a 'triggered by' bug. 
I could not find anything specific to what we need to write in comments in ocs-ci, but the closest one is 
* Both the analysis and corrective measures implementation is done and tracked in JIRA project "OCSQE Close Loop (OCSQECL)" in ODF QE closed loop process slides. I think we need to link them somehow.